### DISCUSSION TITLE
onAuthStateChanged not triggering on sign in and log out

### DISCUSSION THREAD

**Discussion by mypalvikram** [invertase/react-native-firebase#6560] [Q&A] [answered]
Looking for some assistance on why my `onAuthStateChanged` is not triggering when I log out or sign in. If I refresh the application it routes to the proper screens on that first check. I'm currently running React Native 0.70.1 and RNFB 15.6. I'm also using Jotai for state management.

authHandlers.tsx
```
import auth from '@react-native-firebase/auth';

export const handleSignIn = async (email: string, password: string) => {
  try {
    await auth()
      .signInWithEmailAndPassword(email, password)
      .then(() => console.log('User signed in!'));
  } catch (error) {
    console.log(error);
  }
};

export const handleSignOut = async () => {
  try {
    await auth()
      .signOut()
      .then(() => console.log('Signed out!'));
  } catch (error) {
    console.log(error);
  }
};
```

Navigation.tsx
```
export default function Navigation() {
  const [user, setUser] = useAtom(userAtom);
  const [authReady, setAuthReady] = useAtom(authReadyAtom);

  useEffect(() => {
    const subscriber = auth().onAuthStateChanged(
      (user: FirebaseAuthTypes.User | null) => {
        user ? setUser(user) : setUser(null);
      },
    );

    setAuthReady(true);
    console.log('fired'); <-- Only logs on initial load and not when I sign in or sign out

    subscriber();
  }, []);

  if (!authReady) {
    return <LoadingScreen />;
  } else {
    return (
      <NavigationContainer linking={Linking}>
        {user ? <AppStackNavigator /> : <AuthStackNavigator />}
      </NavigationContainer>
    );
  }
}
```

**Comment by mikehardy** [comment, marked as answer]
There are a bunch of ways things can go wrong, and it sounds like you have an issue, not a discussion, however your mention of other modules makes me question if this is a react-native-firebase problem or not. 

We have a fully working auth demo that demonstrates it working

https://github.com/invertase/react-native-firebase-authentication-example/blob/68bd37f8c219f892655b40c73e9818370f368b82/template/src/app/App.tsx#L38

That line fires.

The dependencies are a little out of date, but nothing about it will stop working if I update, my work projects are up to date and they work - onAuthStateChanged works, I was just using it yesterday, and am about to again this morning.

If you can reproduce using that demonstration we have a bug, otherwise there is something going on in your project and I urge you to make a simple - no other dependencies included - App.tsx that is just react-native + react-native-firebase and does an auth with the listener and show it works (or that it doesn't! Then file an issue)

**Reply by mypalvikram** [reply]
Thank you again! Figured out my issue. It was quite trivial, not returning the subscriber in the useEffect.

**Reply by mikehardy** [reply]
Nothing like going for a reproduction to expose the (usually trivial, and I'm speaking of my own issues...) errors :laughing: - glad you found it, good luck with your project

**Comment by FrancisWorld** [comment]
Mine is getting null
